pkg/controllers/allocation/scheduling/scheduler.go

```diff
@@ -147,3 +156,56 @@ func IsSchedulable(pod *v1.Pod, node *v1.Node) bool {
 	// TODO, support node affinity
 	return true
 }
+
+type schedulerMetricsDecorator struct {
+	scheduler                Scheduler
+	scheduleTimeHistogramVec *prometheus.HistogramVec
+}
+
+func DecorateSchedulerMetrics(scheduler Scheduler) Scheduler {
+	scheduleTimeHistogramVec := prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: metrics.KarpenterNamespace,
+			Subsystem: "allocation_controller",
+			Name:      "scheduling_duration_seconds",
+			Help:      "Duration of scheduling process in seconds. Broken down by provisioner and result.",
+			Buckets:   metrics.DurationBuckets(),
+		},
+		[]string{metrics.ProvisionerLabel, metrics.ResultLabel},
+	)
+	crmetrics.Registry.MustRegister(scheduleTimeHistogramVec)
+
+	return &schedulerMetricsDecorator{scheduler: scheduler, scheduleTimeHistogramVec: scheduleTimeHistogramVec}
+}
+
+func (s *schedulerMetricsDecorator) Solve(ctx context.Context, provisioner *v1alpha3.Provisioner, pods []*v1.Pod) ([]*Schedule, error) {
```

Comment:
Given the similarity of all of these histograms, I think we could likely achieve a lot of code-deduplication. Feel free to punt for a future issue.